python/hail/table.py

```diff
@@ -1291,7 +1291,7 @@ def types_compatible(left, right):
                 analyze('Table.index', e, src._row_indices)
 
             is_key = src.key is not None and len(exprs) == len(src.key) and all(
-                expr is key_field for expr, key_field in zip(exprs, self.key.values()))
+                expr is key_field for expr, key_field in zip(exprs, src.key.values()))
```

Comment:
Didn't we already merge this fix? Why is it showing up in the diff?

Reply:
not sure. it's definitely fixed in master.